salt/utils/dns.py

```diff
@@ -95,7 +95,7 @@ def validate(lookup, ref, match=None):
             return [code for code, name in ref.items() if lookup in name][-1]
         else:
             # OrderedDicts only!(?)
-            return ref.keys()[ref.values().index(lookup)]
+            return list(ref)[ref.values().index(lookup)]
```

Comment:
1. Py3 `dict_values` has no `index` method.
2. This looks weird at all. I know it's not your code but...
Please change it to something like `return {v: k for k, v in ref.items()}[lookup]

Reply:
Oh wow is that what this is doing.. lol I didn't spend enough time to try and figure it out. Done